tutorials/intermediate/legend_guide.py

```diff
@@ -110,25 +113,24 @@
 # figure's top right-hand corner instead of the axes' corner, simply specify
 # the corner's location and the coordinate system of that location::
 #
-#     plt.legend(bbox_to_anchor=(1, 1),
-#                bbox_transform=plt.gcf().transFigure)
+#     ax.legend(bbox_to_anchor=(1, 1),
+#                bbox_transform=fig.transFigure)
 #
 # More examples of custom legend placement:
 
-plt.subplot(211)
-plt.plot([1, 2, 3], label="test1")
-plt.plot([3, 2, 1], label="test2")
+fig, (ax1, ax2) = plt.subplots(nrows=2)
+ax1.plot([1, 2, 3], label="test1")
+ax1.plot([3, 2, 1], label="test2")
 
 # Place a legend above this subplot, expanding itself to
 # fully use the given bounding box.
-plt.legend(bbox_to_anchor=(0., 1.02, 1., .102), loc='lower left',
+ax1.legend(bbox_to_anchor=(0., 1.02, 1., .102), loc='lower left',
            ncol=2, mode="expand", borderaxespad=0.)
 
-plt.subplot(223)
-plt.plot([1, 2, 3], label="test1")
-plt.plot([3, 2, 1], label="test2")
-# Place a legend to the right of this smaller subplot.
-plt.legend(bbox_to_anchor=(1.05, 1), loc='upper left', borderaxespad=0.)
+ax2.plot([1, 2, 3], label="test1")
+ax2.plot([3, 2, 1], label="test2")
+# Place a legend on top center  of this subplot.
```

Comment:
I think the point is to be outside on this one. We can replicate the layout with funny `subplot` mixing, using `subplot_mosaic`:
```python
fig, ax_dict = plt.subplot_mosaic([
    ['top', 'top'],
    ['bottom', '-'],
])
```

Reply:
This is actually pretty cool, wasnt aware of that, bc I was struggling with using `fig, ax = plt.subplots()` to get that layout, hence changing the label to the inside. Thx for the tip @QuLogic